tonilopezmr/day 2/ChecksumPartTwo.kt

```diff
@@ -0,0 +1,57 @@
+package day2
+
+import org.junit.Assert.*
+import org.junit.Test
+
+class ChecksumPartTwo {
+
+  fun evenlyDivisibleChecksum(of: Array<Array<Int>>): Int =
+      of.fold(0) { acc, a ->
+        acc + evenlyDivisibleDivision(a.sorted())
+      }
+
+  fun evenlyDivisibleDivision(array: List<Int>): Int {
+    for (a in array) {
+      for (b in array.minus(a)) {
```

Comment:
If you choose to order the array (line 10), you don't need to iterate the whole array in the nested for, you just need to look at the elements on the left of `a`, as those on the right will be `> a` and  `a / b | b > a` will never be a even division.

Reply:
Nice observation 💯

Reply:
👍